src/components/DonationCard/DonationCard.tsx

```diff
@@ -1,23 +1,67 @@
 import styles from './DonationCard.module.css';
 
-export interface DonationCardProps {
+interface DonationCardProps {
    title: string;
    description: string;
    donationTarget: number;
+   amount: number;
+   isOpen: boolean;
+   onShowMore: () => void;
 }
 
-function DonationCard({ title, description, donationTarget }: DonationCardProps) {
+const getButtonClassNames = (progress: number) => {
+   return `${styles.donationCardButton} ${
+      progress < 33 ? styles.low : progress < 67 ? styles.med : styles.high
+   }`;
+};
+
+const getDonationCardProgressClassNames = (progress: number) => {
+   return `${styles.donationCardProgress} ${
+      progress < 33 ? styles.low : progress < 67 ? styles.med : styles.high
+   }`;
+};
+
+function DonationCard({
+   title,
+   description,
+   donationTarget,
+   amount,
+   isOpen,
+   onShowMore,
+}: DonationCardProps) {
+   const progress = (amount / donationTarget) * 100;
+
    return (
-      <article className={styles.donationCard}>
-         <header>
-            <h3>{title}</h3>
+      <article className={styles.donationCard} onClick={onShowMore}>
+         <header className={styles.donationCardHeader}>
+            <h2 className={styles.donationCardTitle}>{title}</h2>
+            <button className={getButtonClassNames(progress)}>
+               Donate Now!
+            </button>
          </header>
-         <section>
-            <p>{description}</p>
-            <footer>
-               <p>Target: {donationTarget}</p>
-            </footer>
-         </section>
+         {isOpen && (
+            <section className={styles.donationCardDescription}>{description}</section>
+         )}
+         <footer>
+            <section className={styles.donationCardTargetInfo}>
+               <div className={styles.donationCardProgressBar}>
+                  <div
+                     className={getDonationCardProgressClassNames(progress)}
+                     style={{
+                        width: `${progress}%`,
+                        pointerEvents: 'none',
+                     }}>
+                     {progress.toFixed(0)}%
+                  </div>
```

Comment:
For example this can be done:
```markdown
const generateProgressClass = (progress: number) => {
   const className = `progressWidth${progress.toFixed(0)}`;
   if (!document.styleSheets[0].rules.namedItem(className)) {
      const style = document.createElement('style');
      style.innerHTML = `
         .${className} {
            width: ${progress}%;
         }
      `;
      document.head.appendChild(style);
   }
   return className;
};
```
or this:
```markdown
const progressBarRef = useRef<HTMLDivElement>(null);
useEffect(() => {
  if (progressBarRef.current) {
     progressBarRef.current.style.width = `${progress}%`;
  }
}, [progress]);
.
.
.
<div
   ref={progressBarRef}
   className={getDonationCardProgressClassNames(progress)}
>
   {progress.toFixed(0)}%
</div>
.
.
.
```
But does it holds any pros?

Reply:
Well this is one of the cases that the inline style IS important and useful.
Changing a size of something dynamically with the importance of the inline-style is a good reason to use it.

So here is a great case showing a good inline-style.


That being said, good article to why a lot of developers do not use it (except for these specific cases) - https://dev.to/alim1496/avoid-using-inline-css-styles-5b6p